Make company descriptions collapsible

PeterJFB opened the pull request:
# Description

Make company descriptions over a certain length collapsible, making the succeeding components more accessible.

# Result

**Before**
![Screenshot from 2023-02-19 14-47-36](https://user-images.githubusercontent.com/42469466/219952398-9b09520b-8faf-48a4-ad9f-0d43363c54ff.png)

**After**
Long description: 
![Screencast from 02-19-2023 02:46:45 PM.webm](https://user-images.githubusercontent.com/42469466/219952359-85a95f3b-0f91-4044-b916-7ca3850798cb.webm)

Forgot to record in light theme, but here's how that looks^
![image](https://user-images.githubusercontent.com/42469466/219966107-ff69758b-1a79-4bde-9823-4b2738392b11.png)


Short descriptions will not have the option to expand the content
![Screenshot from 2023-02-19 14-48-17](https://user-images.githubusercontent.com/42469466/219952401-d8a786e6-ea41-4e91-8042-c46bb9e31c44.png)

# Testing

- [x] I have thoroughly tested my changes.

Ensured that the component does not perform any unnecessary animation while loading into the page with short or long content. Made sure the interactions works nicely on mobile as well:)

---

Resolves [ABA-294](https://linear.app/abakus-webkom/issue/ABA-294)

ivarnakken commented on app/components/CollapsibleDisplayContent/CollapsibleDisplayContent.css:
```diff
@@ -0,0 +1,43 @@
+.collapse {
+  position: relative;
+  overflow: hidden;
+  transition: 0.5s var(--easing) height;
```
Use `transition: height var(--easing-slow) instead`, please. It's important to not use different speeds everywhere.

ivarnakken commented on app/components/CollapsibleDisplayContent/index.tsx:
```diff
@@ -0,0 +1,92 @@
+import styles from './CollapsibleDisplayContent.css';
+
+type Props = {
+  /** The content to be displayed - the text */
+  content: string;
+
+  /** The id of the div wrapping the content - the id */
+  id?: string;
+
+  /** The classname of the div wrapping the content - the className */
+  className?: string;
+
+  /** Any style tp be added to the div wrapping the content - the style */
```
Although comments are often nice, I'm more a fan of not commenting on props, as I feel they are self explanatory. If you need to explain the props, that just means the naming is bad.

Also there's a typo on the last one; `tp -> to`.

PeterJFB replied:
I agree. Removed them from `DisplayContent as well`

ivarnakken commented on app/components/CollapsibleDisplayContent/index.tsx:
```diff
@@ -0,0 +1,92 @@
+/**
+ * Renders `content` produced by the Editor component in a read-only format.
+ */
+function CollapsibleDisplayContent({
```
I'd prefer a short name, like `CollapsibleContent`. Not sure why "display" is needed (I get that it was named that from before).

PeterJFB replied:
I understand the use of `Display` as the content being "read-only", though there might exist better names for it ...

ivarnakken replied:
I'm not following you on the "read-only" part. `CollapsibleContent` is much cleaner and more precise.

PeterJFB replied:
It makes more sense if you look at how it is implemented: It uses the `Editor`-component to display the content, but since you cannot edit it, it is only a "display".

I'd prefer keeping the name as it is such that it is easy to see it is related to the `DisplayContent` and not the `Collapse`-wrapper (which we also have but isn't flexible enough for this purpose unfortunately).

ivarnakken commented on app/components/CollapsibleDisplayContent/index.tsx:
```diff
@@ -0,0 +1,92 @@
+import Editor from '@webkom/lego-editor';
+import '@webkom/lego-editor/dist/style.css';
+import 'react-image-crop/dist/ReactCrop.css';
```
Why do these two need to be imported?

PeterJFB replied:
Yikes, no. Thanks for spotting :eyes:

PeterJFB replied:
Yikes yes***. Turns out they add styling-properties to the lego-editor :sweat_smile:

ivarnakken commented on app/components/CollapsibleDisplayContent/index.tsx:
```diff
@@ -0,0 +1,92 @@
+import '@webkom/lego-editor/dist/style.css';
+import 'react-image-crop/dist/ReactCrop.css';
+import { type CSSProperties, useRef, useState } from 'react';
+import Icon from '../Icon';
```
Absolute import, please. I'm sure it was just an auto import though.

ivarnakken reviewed (changes requested):
Very cool! 💯💯 

I have some minor comments, but they aren't that important. What bothers me more is the trigger icon on top of the text. Since it is outlined, it sort of blends in together with the text, and I'm not a big fan of it. With "show more / expanding content triggers" I prefer flat text buttons, like these:

<img width="444" alt="image" src="https://user-images.githubusercontent.com/69514187/219954572-ae63b466-0d7c-4e11-a492-18f2ba7602fb.png">

But, whatever you end up with, please don't put it on top of the text.

LudvigHz commented:
Nice! :100: 
But why not just add a `collapse` prop on displayContent instead of making a nearly identical one?

PeterJFB commented:
> Very cool! 100100
> 
> I have some minor comments, but they aren't that important. What bothers me more is the trigger icon on top of the text. Since it is outlined, it sort of blends in together with the text, and I'm not a big fan of it. With "show more / expanding content triggers" I prefer flat text buttons, like these:
> 
> <img alt="image" width="444" src="https://user-images.githubusercontent.com/69514187/219954572-ae63b466-0d7c-4e11-a492-18f2ba7602fb.png">
> 
> But, whatever you end up with, please don't put it on top of the text. I like the "fade" a lot though!

Yeah i agree the outline was clashing with the text too much. 

The updated look is on the light theme picture. I moved it and added a smooth background to it, think it looks way cleaner now^

PeterJFB commented:
> Nice! 100 But why not just add a `collapse` prop on displayContent instead of making a nearly identical one?

Honest answer, I was lazy:P

Looking over the other pages, I didn't really think it was worth the time to make the more logic-intensive component. I also think this solution will have the least performance-impact, since it will only add the `UseRef`-code to components that don't need it.

ivarnakken reviewed (approved):
:shipit:

Arashfa0301 reviewed (approved):
Looks good to me :)

ingraso reviewed (approved):
Nice work! Looks very nice :rocket:
